### ISSUE TITLE
Issue about the attacked graph

### ISSUE THREAD

**Issue by NozomiSG** [zknus/NeurIPS-2023-HANG-Robustness#2]
Thanks for sharing the code. I'd like to ask you some questions about the attacked graph.

When I was trying to reproduce the results of Table 4 in your paper, I realized that the file of the perturbed graph loaded in line 136 of your code run_metattack_rate.py is located locally on your computer. When I tried to test it with citeseer dataset, I found that the results obtained were not as expected, with multiple predictions close to 67%.

Could you provide your perturbated graph files or explain the potential  causes of this results? Looking forward to your reply.

**Comment by NozomiSG** [comment]
I followed your procedures on cora and citeseer datasets and got the following results. Could you please tell me if this result is correct? Looking forward to your reply!
<img width="516" alt="Screenshot 2024-04-01 at 6 04 21 PM" src="https://github.com/zknus/NeurIPS-2023-HANG-Robustness/assets/75309578/15e973a0-a116-4c62-a581-a8c0116aeea5">

**Comment by zhongjian-zhang** [comment]
I am also meet the same question, have you solved it yet?

**Comment by zhongjian-zhang** [comment]
@zknus If you have time, could you please answer it?

**Comment by zknus** [comment]
Hi @1957787636 and @NozomiSG ,

Thank you both for your diligence on this project. Apologies for my delayed response; I did not receive a notification from GitHub regarding this issue.

Regarding Table 4, please note that the datasets for Cora and Citeseer were not included. Based on your findings, the HANG model continues to perform robustly, though it does require further fine-tuning to improve its clean accuracy. We have updated and uploaded the run_metattack_rate.py script, which now allows direct loading of the perturbed graph used in Table 4.

For your information, the perturbed graphs were generated using the script available at: https://github.com/DSE-MSU/DeepRobust/blob/master/examples/graph/test_mettack.py  

Thank you once again for your contributions and attention to detail.

**Comment by zhongjian-zhang** [comment]
> Hi @1957787636 and @NozomiSG ,
> 
> Thank you both for your diligence on this project. Apologies for my delayed response; I did not receive a notification from GitHub regarding this issue.
> 
> Regarding Table 4, please note that the datasets for Cora and Citeseer were not included. Based on your findings, the HANG model continues to perform robustly, though it does require further fine-tuning to improve its clean accuracy. We have updated and uploaded the run_metattack_rate.py script, which now allows direct loading of the perturbed graph used in Table 4.
> 
> For your information, the perturbed graphs were generated using the script available at: https://github.com/DSE-MSU/DeepRobust/blob/master/examples/graph/test_mettack.py
> 
> Thank you once again for your contributions and attention to detail.

Thank you for the response! I tried tuning some hyperparameters of HANG, but it didn't improve HANG's performance on less perturbation rate or even clean graphs. The acc of HANG is far below vanilla gcn in these case.

**Comment by 1957787636** [comment]
I again tune the hyperparameter. Finally, the results seem reasonable when I run the following command:

`python run_metattack_rate.py --dataset cora --function hang --block constant --lr 0.005 --dropout 0.6 --input_dropout 0.8 --time 25 --hidden_dim 64 --step_size 1 --runtime 10 --epochs 800 --patience 150`
